### ISSUE TITLE
Directory support on Virtual Drive Z: and major cleanups on this drive - bad implementation!

### ISSUE THREAD

**Issue by prino** [joncampbell123/dosbox-x#2587]
Please revert this change or make it internally transparent, i.e. automagically search the new directories as if they were part of the path, as I suddenly find that commands which I use all the time (MOUNT/FIND/RESCAN) now result in 

'Bad command or filename - "[cmd]"'

Copying commands from the virtual subdirectories to z:\ is also not directly possible,

"copy z:\bin\mount.com z:\" 

results in a 

"Insufficient disk space - MOUNT,COM"
0 File(s) copied.

message! Or could  this be because z:\ is mounted R/O?

Doing it via a MOUNT created drive works, but not from within DOSBox-X, you have to use Explorer to copy it from the mounted-as-drive directory back to the "drivez" DOSBox-X directory.

**Comment by Wengier** [comment]
No, it is not because of "bad" implementation, but it is simply a %PATH% setting issue. The %PATH% environment variables in DOSBox-X now includes various directories in the Z drive, but if you use a custom %PATH% setting, please make sure to include them yourself. This can be done very easily - just check the setting of the %PATH% environment variable.

Moreover, the Z drive is still read-only, so you cannot copy files to the drive directly. This is same as before. Please copy needed files to the "drivez" directory for now.

**Comment by prino** [comment]
My sincere apologies!

You could have mentioned something about this in the changelog, the only time I ever started the program with a "virgin" (at the time just containing "Z:\") path was the time I installed it, and added the mounted disks, and it's likely that others have done the same. It may not standard DOS behaviour, but it would be nice if these 6 new directories would automagically be searched, even if the path only contains Z:\ - after all "LS.COM" and many of its siblings aren't really .COM files either, but they do run as if they are...

**Comment by Wengier** [comment]
@prino I have already mentioned this in the changelog. As for the %PATH%, I think it is likely better to have an option (enabled by default for now) to automatically expand the %PATH% environment variable to include these directories when it is set and only contains "Z:\" (perhaps in addition to paths in other drives) rather than automatically searching for all directories on Z drive even if %PATH% does not contain them. Please check out the updated Windows builds:

* Windows SDL1 build: [dosbox-x-update.zip](http://cndos.fam.cx/dosware/dosbox-x-update.zip)
* Windows SDL2 build: [dosbox-x-sdl2-update.zip](http://cndos.fam.cx/dosware/dosbox-x-sdl2-update.zip)

**Comment by prino** [comment]
Thanks for the update, downloaded the update above, but it's a 32-bit version, while I'm running the 64-bit version. Will patiently wait until the end of the month, to much other things to do anyway, tax, garden, lading car, and driving back to Lithuania...

**Comment by Wengier** [comment]
@prino You can download the 64-bit MinGW SDL1 build from:

* [dosbox-x-mingw64.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64.zip)

Hope this helps.

**Comment by prino** [comment]
Works like a charm in 0.83.15. However, it does create a very long path that may, what was the real DOS limit, conflict with real paths.

**Comment by Wengier** [comment]
@prino DOSBox-X supports directories in LFN forms in %PATH% too, and I think paths containing Z subdirectories are generally shorter than paths with LFNs.
